### PR TITLE
ci: add helm install tests

### PR DESCRIPTION
**Pull request by jedevc**
This will help avoid issues like https://github.com/dagger/dagger/issues/7916 in the future - this took a lot of messing around to fix, so it would be good to *never* hit this again.

### CODE REVIEW

**Comment by jedevc** [comment]
*Hm* - there appears to be something in our CI infra here:
- Using our dagger runners (fail): https://github.com/dagger/dagger/actions/runs/10285227388/job/28462983902?pr=8116
  `RunContainerError: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error setting cgroup config for procHooks process: failed to write "a *:* rwm": write /sys/fs/cgroup/devices/kubepods/besteffort/podac92bc34-526c-43c7-af5a-a80e9fedc5c0/8ed7faec5bc845ae4599a250fdd3d1e6b982d221e018155101ff1de0ac1be9b4/devices.allow: operation not permitted: unknown`
- Using `ubuntu-latest` (success): https://github.com/dagger/dagger/actions/runs/10286345815/job/28466773136?pr=8116

Not 100% sure what would be causing this, cc @matipan @gerhard, any ideas?

**Comment by gerhard** [comment]
I suspect that we need `cgroup v2` for nesting support for: https://github.com/marcosnils/daggerverse/blob/e0bd6b9f5519c49db4b6eb0689927214720976f9/k3s/main.go#L13-L36

To get cgroup v2, we would need to switch from AL2 to either:
- `AL2023` https://github.com/aws/containers-roadmap/issues/1953
- or `Bottlerocket`

We have been talking about this with @matipan a few times in the past. I propose that we create a new experimental runner group which uses this new OS. WDYT @matipan ?

**Comment by jedevc** [comment]
Thanks @gerhard :tada: If that's going to take a bit of time, is there any reason we can't switch these jobs to use `ubuntu-latest`, and later change them back?

At least that way we can get the helm coverage asap!

**Comment by gerhard** [comment]
Yes, `ubuntu-latest` sounds like a good short-term solution 👍

**Comment by jedevc** [comment]
Rebased, and updated to use `ubuntu-latest` - I think this should be alright to merge now? cc @gerhard @marcosnils

File: .dagger/helm.go [hunk 1]
```diff
@@ -2,37 +2,94 @@ package main
...
+
+	test, err := h.chart().
+		WithMountedFile("/usr/bin/dagger", cli).
+		WithServiceBinding("helm-test", k3ssvc).
+		WithFile("/.kube/config", k3s.Config()).
+		WithEnvVariable("KUBECONFIG", "/.kube/config").
+		WithEnvVariable("CACHEBUSTER", identity.NewID()).
+		WithExec([]string{"kubectl", "get", "nodes"}).
+		WithExec([]string{"helm", "install", "--create-namespace", "--namespace=dagger", "dagger", "."}).
+		WithExec([]string{"kubectl", "wait", "--for=condition=Ready", "--timeout=120s", "pod", "--selector=name=dagger-dagger-helm-engine", "--namespace=dagger"}).
```
Comment (marcosnils) [inline .dagger/helm.go:52]: nit: this can be removed by adding `--wait` to the `helm install` command.

**Review by marcosnils — APPROVED** [review]

**Pull request merged**